gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -373,6 +382,18 @@ public GoogleCloudStorageImpl(
             ? null
             : this.storage.getRequestFactory().getInitializer();
 
+    if (this.storageOptions.isMetricsEnabled()) {
+      checkArgument(
+          httpRequestInitializer instanceof RetryHttpInitializer,
+          "request initializer must be an instance of the RetryHttpInitializer class"
+              + " when gRPC API enabled");
+      Credentials credentials = ((RetryHttpInitializer) httpRequestInitializer).getCredentials();
+      this.metricsPublisher =
+          CloudMonitoringMetricsPublisher.create(options.getProjectId(), credentials);
+    } else {
+      this.metricsPublisher = new NoOpMetricsPublisher();
+    }
```

Comment:
nit: consider using ternary operator, but not sure if it will make code more readable in this specific case.